src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java

```diff
@@ -166,12 +165,12 @@ else if (httpCode == 409 && e.getMessage().contains("Operation cannot be fulfill
             ObjectMeta podMetadata = pod.getMetadata();
             template.getWorkspaceVolume().createVolume(client, podMetadata);
             template.getVolumes().forEach(volume -> volume.createVolume(client, podMetadata));
-            watcher = new AllContainersRunningPodWatcher(client, pod);
-            try (Watch w1 = client.pods().inNamespace(namespace).withName(podName).watch(watcher);
-                 Watch w2 = eventWatch(client, podName, namespace, runListener)) {
-                assert watcher != null; // assigned 3 lines above
-                watcher.await(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+
+            Pod checkPod = client.pods().inNamespace(namespace).withName(podName).waitUntilReady(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
+            if (!Readiness.isPodReady(checkPod)) {
+                throw new KubernetesClientTimeoutException(pod, template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
             }
```

Comment:
Did we establish whether `waitUntilReady` throws an exception if it times out, in which case
```suggestion
            client.pods().inNamespace(namespace).withName(podName).waitUntilReady(template.getSlaveConnectTimeout(), TimeUnit.SECONDS);
```
?